Taskfile.yml

```
@@ -1011,6 +1026,20 @@ tasks:
     status:
       - test -x {{.BIN_DIR}}/uv
 
+  deps:cosign:
+    desc: Install cosign
+    internal: true
+    vars:
+      ARCH_TYPE: '{{ if eq ARCH "amd64" }}amd64{{ else }}{{ARCH}}{{ end }}'
+      BINARY_NAME: 'cosign-{{OS}}-{{.ARCH_TYPE}}'
+    cmds:
+      - cmd: echo "Downloading Cosign v{{.COSIGN_VERSION}}..."
+      - cmd: curl -O -L "https://github.com/sigstore/cosign/releases/download/v{{.COSIGN_VERSION}}/{{.BINARY_NAME}}"
+      - cmd: sudo mv {{.BINARY_NAME}} /usr/local/bin/cosign
+      - cmd: sudo chmod +x /usr/local/bin/cosign
```

nits: cleanup and removal of sudo
TODO: please try to find a way to make this work without adding this to user installation

```suggestion
      - echo "Downloading Cosign v{{.COSIGN_VERSION}}..."
      - curl -O -L "https://github.com/sigstore/cosign/releases/download/v{{.COSIGN_VERSION}}/{{.BINARY_NAME}}"
      - mv {{.BINARY_NAME}} /usr/local/bin/cosign
      - chmod +x /usr/local/bin/cosign
```

---

an alternative would be to add cosign as a requirement for running cosign locally. this is not a requirement when running other installation types like kubernetes.